legacy/firmware/ethereum.c

```diff
@@ -1042,26 +1052,27 @@ bool ethereum_parse(const char *address, uint8_t pubkeyhash[20]) {
 }
 
 bool ethereum_path_check(uint32_t address_n_count, const uint32_t *address_n,
-                         bool pubkey_export, uint64_t chain) {
+                         bool pubkey_export,
+                         const EthereumDefinitionsDecoded *defs) {
   bool valid = (address_n_count >= 3);
   valid = valid && (address_n[0] == (PATH_HARDENED | 44));
   valid = valid && (address_n[1] & PATH_HARDENED);
   valid = valid && (address_n[2] & PATH_HARDENED);
   valid = valid && ((address_n[2] & PATH_UNHARDEN_MASK) <= PATH_MAX_ACCOUNT);
 
   uint32_t path_slip44 = address_n[1] & PATH_UNHARDEN_MASK;
-  if (chain == CHAIN_ID_UNKNOWN) {
+  if (!defs || defs->network.chain_id == CHAIN_ID_UNKNOWN) {
```

Comment:
if the chain_id is unknown, that means network not resolved, that means _it is not in the builtin list_, that means `is_ethereum_slip44` definitely returns false.

this outher if should go away and this usecase folds into `slip44 == SLIP44_UNKNOWN` below

Reply:
9556b8f18d0dab6fc84e20f31bad8ba707737562